central/processbaseline/datastore/datastore.go

```diff
@@ -26,20 +27,24 @@ type DataStore interface {
 	RemoveProcessBaselinesByDeployment(ctx context.Context, deploymentID string) error
 	RemoveProcessBaselinesByIDs(ctx context.Context, ids []string) error
 	UpdateProcessBaselineElements(ctx context.Context, key *storage.ProcessBaselineKey, addElements []*storage.BaselineItem, removeElements []*storage.BaselineItem, auto bool) (*storage.ProcessBaseline, error)
-	UpsertProcessBaseline(ctx context.Context, key *storage.ProcessBaselineKey, addElements []*storage.BaselineItem, auto bool) (*storage.ProcessBaseline, error)
+	UpsertProcessBaseline(ctx context.Context, key *storage.ProcessBaselineKey, addElements []*storage.BaselineItem, auto bool, lock bool) (*storage.ProcessBaseline, error)
 	UserLockProcessBaseline(ctx context.Context, key *storage.ProcessBaselineKey, locked bool) (*storage.ProcessBaseline, error)
 
 	WalkAll(ctx context.Context, fn func(baseline *storage.ProcessBaseline) error) error
+
+	BuildUnlockedProcessBaseline(ctx context.Context, key *storage.ProcessBaselineKey) (*storage.ProcessBaseline, error)
+	ClearProcessBaseline(ctx context.Context, key *storage.ProcessBaselineKey, lock bool) (*storage.ProcessBaseline, error)
```

Comment:
It seems `lock option` is not necessary? 

Also please add comments to these two new functions if needed, it may contain some side effect that is not explicit.